server/models/User.js

```diff
@@ -21,9 +22,9 @@ const userSchema = new mongoose.Schema({
     default: Date.now,
   },
   profile: {
-    type: mongoose.Schema.Types.ObjectId,
+    type: Profile.schema,
```

Comment:
I think it's better to reference the profile model here (store only the object id of the profile document), storing the schema within the profile field is similar to embedding the entire document here